great_expectations/execution_engine/bundled_metric_configuration.py

```diff
@@ -0,0 +1,27 @@
+from dataclasses import asdict, dataclass
+from typing import Any
+
+from great_expectations.core.util import convert_to_json_serializable
+from great_expectations.types import DictDot
+from great_expectations.validator.metric_configuration import MetricConfiguration
+
+
+@dataclass(frozen=True)
+class BundledMetricConfiguration(DictDot):
+    """
+    BundledMetricConfiguration is a "dataclass" object, which holds components required for bundling metric computation.
+    """
+
+    metric_configuration: MetricConfiguration
+    metric_fn: Any
```

Comment:
If this is meant to contain a function this should probably be  `typing.Callable`?
`from typing import Callable`

```suggestion
    metric_fn: Callable
```
You could make it fancier and declare the specific args and return type the callable expects it just leave it as a plain `metric_fn: Callable`.
https://docs.python.org/3/library/typing.html#typing.Callable

Reply:
@Kilo59 This is not always "Callable"; sometimes it is one of many SQLAlchemy functions, too numerous to specify (by the way, we need to figure out how to handle such situations regarding your on-going work in promoting stating type hints universally).  Also, this PR is in the draft stage (I am battling a bug; once fixed, it will be reviewable).  Thanks!